internal/provider/resource_settings.go

```diff
@@ -0,0 +1,136 @@
+created with the organization. When a Terraform resource is created, settings automatically
+imported while no action is taken when the resource is deleted.
+
+-> This resource requires Infra server version 0.20.0 or higher.
+`,
+
+		CreateContext: resourceSettingsUpdate,
+		ReadContext:   resourceSettingsRead,
+		UpdateContext: resourceSettingsUpdate,
+		DeleteContext: resourceSettingsDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"password_requirements": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"minimum_length": {
+							Description: "Minimum password length.",
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Default:     8,
+							ValidateDiagFunc: validation.ToDiagFunc(
+								validation.IntAtLeast(8),
+							),
+						},
+						"minimum_lowercase": {
+							Description: "Minimum number of lowercase ASCII letters.",
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Default:     0,
+						},
+						"minimum_uppercase": {
+							Description: "Minimum number of uppercase ASCII letters.",
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Default:     0,
+						},
+						"minimum_numbers": {
+							Description: "Minimum number of numbers.",
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Default:     0,
+						},
+						"minimum_symbols": {
+							Description: "Minimum number of symbols.",
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Default:     0,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceSettingsRead(ctx context.Context, d *schema.ResourceData, m any) diag.Diagnostics {
+	client := m.(*api.Client)
+
+	settings, err := client.GetSettings(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	requirements := map[string]int{
+		"minimum_length":    settings.PasswordRequirements.LengthMin,
+		"minimum_lowercase": settings.PasswordRequirements.LowercaseMin,
+		"minimum_uppercase": settings.PasswordRequirements.UppercaseMin,
+		"minimum_numbers":   settings.PasswordRequirements.NumberMin,
+		"minimum_symbols":   settings.PasswordRequirements.SymbolMin,
+	}
+
+	if err := d.Set("password_requirements", []map[string]int{requirements}); err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId("self")
+
+	var diags diag.Diagnostics
+	return diags
+}
+
+func resourceSettingsUpdate(ctx context.Context, d *schema.ResourceData, m any) diag.Diagnostics {
+	client := m.(*api.Client)
+
+	if err := requireMinimumServerVersion(ctx, client, "0.20.0"); err != nil {
+		return diag.FromErr(err)
+	}
+
+	var requirements api.PasswordRequirements
+
+	for i := range d.Get("password_requirements").([]interface{}) {
```

Comment:
This sets password requirements for many orgs? Not sure what the range is

Reply:
It's a way to access the schema of nested objects since `d.Get()` takes a list index. The alternative is to `d.Get("password_requirements")`, check length, type assert items to `map[string]interface{}`, then get individual values. Since the schema specifies `MaxItems:1`, there can only be a single `password_requirements` per settings resource.